src/controllers/order.controller.ts

```diff
@@ -120,4 +125,54 @@ export class OrderController {
       }
     },
   );
+
+  public getUserOrderDetail = asyncHandler(
+    async(req: Request, res: Response) => {
+      const user = (req.session as CustomSessionData).user;
+      if(user!==undefined){
+        const id = parseInt(req.params.id);
+        const order = await this.orderService.getOrderById(id);
+        if(order === null){
+          return res.render('errors/404');
+        }
+        const payment = await this.paymentService.findOrderPayment(id);
+        return res.render('user/order/detail', {
+          currentSite: UserNavBar.ORDER, 
+          user, 
+          order,
+          payment,
+          formatDate,
+        });
+      }else{
+        return res.redirect('/auth/login');
+      }
+    },
+  );
+
+  public updateOrderStatus = asyncHandler(
+    async(req: Request, res: Response) => {
+      const user = (req.session as CustomSessionData).user;
+      if(user!==undefined){
+        const updateOptions = plainToClass(UpdateOrderStatusDto, req.body);
+        const rawErrors = await validate(updateOptions);
+        if( rawErrors.length > 0 ){
+          res.status(400).send({
+            success: false,
+            message: t(Error.BAD_INPUT),
+          });
+        } else {
+          try {
+            await this.orderService.updateOrderStatus(updateOptions);
```

Comment:
thêm xử lý tìm order trc khi update e nha

Reply:
em có xử lý tìm order ở trong service `updateOrderStatus` và sẽ quăng lỗi khi không tìm thấy order ạ:
`const order = await this.orderRepository.findOne({where:{
      id: updateOption.id,
    }});
    if(!order){
      throw new Error(ErrorMessage.BAD_INPUT);
    }`